src/emqx_crl_cache.erl

```diff
@@ -54,9 +54,12 @@
 %%--------------------------------------------------------------------
 
 start_link() ->
-    URLs = emqx:get_env(crl_cache_urls, []),
-    RefreshIntervalMS = emqx:get_env(crl_cache_refresh_interval,
-                                     timer:minutes(15)),
+    Listeners = emqx:get_env(listeners, []),
+    URLs = collect_urls(Listeners),
```

Comment:
this should be done periodically ?
if a user start a listener from settings page (hot-config)

to support listener stop, we'd need a reference count? if yes, we can add it later.

Reply:
I'm not yet familiar with the hot-config, but if it's possible to have something like a post-update hook, then the `emqx_ocsp_cache:register_listener` and `emqx_crl_cache:refresh` APIs could be used to add URLs.  Otherwise, a timer could be a simpler and good idea.  :smiley_cat:

Reply:
but `emqx_crl_cache:refresh` does not collect new urls from new listner.

Reply:
also it's better to be a gen_cast.

Reply:
> but `emqx_crl_cache:refresh` does not collect new urls from new listner.

It receives an URL as argument, so the new listener could call it with its URLs:

https://github.com/emqx/emqx/blob/ea31fc4529e9c3f70a2e7f9a9bbebb9f4ffcac4b/src/emqx_crl_cache.erl#L65-L66

> also it's better to be a gen_cast.

Ok, I'll change it now.